### DISCUSSION TITLE
Is it possible to transcribe podcasts by URL?

### DISCUSSION THREAD

**Discussion by fearby** [danielmiessler/Fabric#672] [Q&A]
Is this likely to be possible - using ts can do it if I download a podcast as an mp3 but would be good if it was possible to extract summaries like in YouTube.

**Comment by xeniode** [comment]
You would need some tool to grab the audio at least. I have integrated `yt-dlp` into fabric for much more functionality with youtube videos. If you use that, you can grab audio and pipe it through ts. 

`yt-dlp -x --audio-format mp3 https://www.youtube.com/watch?v=dQw4w9WgXcQ`

**Reply by djl0** [reply]
Hey, curious what you've done with yt-dlp. I was looking to use that instead of the official youtube API. I noticed the `yt` command directly calls the transcript endpoint. You mentioned downloading audio and running through `ts`; do you think this is preferable to just using yt-dlp to download subs (described [here](https://www.reddit.com/r/youtubedl/comments/15fcrmd/transcript_extract_from_youtube_videos_ytdlp/))?

Thanks

**Reply by xeniode** [reply]
O yea for sure. I should have mentioned that. I do run the subs directly (if they are available). You can write a small script that will check for subs, and if no subs, grab the audio instead. I have found most videos will only have auto generated subs, if they have any.

**Reply by djl0** [reply]
Just to clarify, the fabric-official yt tool only uses the youtube-provided transcript, right? https://github.com/danielmiessler/fabric/blob/d75ea473e64c387c69fb59ad021bd38f8b507ae3/installer/client/cli/yt.py#L98

How do you tell if it's auto-generated or user-provided transcript? And are you saying that the fabric transcription would be better than youtube's autogenerated?

Thanks

**Reply by fearby** [reply]
> You would need some tool to grab the audio at least. I have integrated `yt-dlp` into fabric for much more functionality with youtube videos. If you use that, you can grab audio and pipe it through ts.
> 
> `yt-dlp -x --audio-format mp3 https://www.youtube.com/watch?v=dQw4w9WgXcQ`

Can I ask how you did this? Thanks

**Reply by djl0** [reply]
@fearby if the podcast is on youtube, you can use the command Xeniode posted above. If not, you need the mp3 link from either rss feed or the webpage or something. After downloading the audio, you would use the `ts` command in this repo to transcribe the audio to text. then pipe that text into fabric.

If the podcast is on youtube, i'm generally fine with the youtube transcription (which I imagine is lesser quality), so i just pull the transcript itself `yt-dlp --quiet --no-warnings --skip-download --write-subs --write-auto-subs --sub-lang en --sub-format ttml --convert-subs srt --output "%(title)s-%(uploader)s--%(upload_date>%Y-%m-%d)s.%(ext)s" --restrict-filenames <youtube video url or id>` then i use sed to clean up the srt file (i can share that if you're going that route, but i don't want to crowd this up too much).

**Reply by ujconsulting** [reply]
> You would need some tool to grab the audio at least. I have integrated `yt-dlp` into fabric for much more functionality with youtube videos. If you use that, you can grab audio and pipe it through ts.
> 
> `yt-dlp -x --audio-format mp3 https://www.youtube.com/watch?v=dQw4w9WgXcQ`
how did you install/ integrate yt-dlp into fabric? For Example how did i get path plus filename to the ts script as an option, so that it downloads the audiofile, passes it to ts, transcribe it and save the transcript somewhere

**Reply by drlakshan** [reply]
> @fearby if the podcast is on youtube, you can use the command Xeniode posted above. If not, you need the mp3 link from either rss feed or the webpage or something. After downloading the audio, you would use the `ts` command in this repo to transcribe the audio to text. then pipe that text into fabric.
> 
> If the podcast is on youtube, i'm generally fine with the youtube transcription (which I imagine is lesser quality), so i just pull the transcript itself `yt-dlp --quiet --no-warnings --skip-download --write-subs --write-auto-subs --sub-lang en --sub-format ttml --convert-subs srt --output "%(title)s-%(uploader)s--%(upload_date>%Y-%m-%d)s.%(ext)s" --restrict-filenames <youtube video url or id>` then i use sed to clean up the srt file (i can share that if you're going that route, but i don't want to crowd this up too much).

HI there can you please teach me how to use ts in fabric? Thanks

**Comment by nic0711** [comment]
How about a link to an Apple Podcast (they normaly have a transcription) or a direct MP3 file link from podigee.io?
